src/cupbearer/detectors/statistical/mahalanobis_detector.py

```diff
@@ -1,19 +1,61 @@
+import numpy as np
 import torch
+from einops import rearrange
 
 from cupbearer.detectors.statistical.helpers import mahalanobis
 from cupbearer.detectors.statistical.statistical import (
     ActivationCovarianceBasedDetector,
 )
 
 
+def _pinv(C, rcond):
+    # Workaround for pinv not being supported on MPS
+    if C.device.type == "mps":
+        return torch.linalg.pinv(C.cpu(), rcond=rcond, hermitian=True).to(C.device)
+    return torch.linalg.pinv(C, rcond=rcond, hermitian=True)
```

Comment:
Alternatively https://lightning.ai/docs/pytorch/stable/accelerators/mps_basic.html#mps-training-basic or calculating pinv from eigh (if torch.linalg.eigh is supported on mps). But, I have no real objections against this either.

Reply:
Probably we should be using https://pytorch.org/docs/stable/generated/torch.linalg.lstsq.html (or at least something other than inverting the matrix), will see if that works